AutomatticTracks/src/main/java/com/automattic/android/tracks/TracksClient.java

```diff
@@ -108,8 +110,11 @@ public void run() {
                     if (shadowCopyEventList.size() > 0) {
                         //2.  get the lock over the DB and write data
                         synchronized (mDbLock) {
-                            for (Event currentEvent : shadowCopyEventList) {
-                                EventTable.insertEvent(mContext, currentEvent);
+                            // do not write events if the queue is already full
+                            if (EventTable.getEventsCount(mContext) < DEFAULT_EVENTS_QUEUE_MAX_SIZE) {
```

Comment:
👋 @daniloercoli . If I understand this correctly, the code will stop logging newer events if the queue is full. Have you perhaps dismissed the alternative of ejecting older events in favor of the newer ones? Especially with the added 14-day rule, keeping the older instead of the newer can get more events closer to that 14-day drop threshold.

Reply:
> Have you perhaps dismissed the alternative of ejecting older events in favor of the newer ones?

Not dismissed the idea at all, just wanted to keep the code simple for now, and dismissed new events only. It's not a common situation that the queue is full, in fact the issue was only reported on iOS since now. That was probably due to a bug in the iOS code, since we're sending events to the server as soon as there will be > 10 in the queue.

Reply:
Maybe we should add a comment there about this?

Reply:
I think we can keep it as is then regarding the full queue policy.

Also, a comment can never hurt I guess so, 👍 for adding one.

Taking this opportunity to hash out a simple (perhaps) way out (but not pushing for it on this PR): wdyt about flushing the queue completely when it's full and a new event comes in? We could also inject a new event to track that we forcefully flushed the queue to accommodate for new events.

Reply:
> Taking this opportunity to hash out a simple (perhaps) way out (but not pushing for it on this PR): wdyt about flushing the queue completely when it's full and a new event comes in? We could also inject a new event to track that we forcefully flushed the queue to accommodate for new events.

This is already done in line 119 - `mDbLock.notifyAll();` - that is called regardless if new events are added or not. That code does trigger a chain of events that send all the events in the queue to the server.

Reply:
Oh, interesting. In that case I'm a bit puzzled: if we're flushing the queue one way or the other, why not let the new event be logged?

Reply:
We don't know if the network request, that is made immediately after, will finish without problems. In case of network errors (also when the device is offline) the flush is called, but does not send events to the server, and the queue is not cleared.

Reply:
I see, thanks for expanding!